src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -99,11 +106,16 @@ public RecommendationRequest postRecommendationRequests(
         //get current date right now and set status to pending
         CurrentUser currentUser = getCurrentUser();
         RecommendationRequest recommendationRequest = new RecommendationRequest();
+        if (!recommendationType.equals("Other")) {
+            requestTypeRepository.findByRequestType(recommendationType).orElseThrow(() -> new EntityNotFoundException(RequestType.class, recommendationType));
+            recommendationRequest.setRecommendationType(recommendationType);
+            recommendationRequest.setDetails(details);
+        } else {
+            recommendationRequest.setRecommendationType(details);
+        }
```

Comment:
```suggestion
        requestTypeRepository.findByRequestType(recommendationType).orElseThrow(() -> new EntityNotFoundException(RequestType.class, recommendationType));
        recommendationRequest.setRecommendationType(recommendationType);
        recommendationRequest.setDetails(details);   
```

We aren't wanting to put all of the details into the recommendation request type field, though I can see how you got that from my explanation.

What I meant was, say if you are asking for a recommendation for a Nobel Peace Prize, you'd select "Other" and "Other" would be stored in the recommendation type.

And then, the user would put something like "I'm asking for a recommendation for the Nobel Peace Prize for my work on helping to resolve dispute between people sitting on curbs and the IV Foot Patrol"

Reply:
Ah I see. I will change that